src/app/shared/conditional-show/conditional-show.model.ts

```diff
@@ -30,4 +32,18 @@ export class ShowCondition {
   private removeStarChar(s: string) {
     return s.substring(0, s.length - 1);
   }
+
+  matchByCaseFields(caseFields: CaseField[]): boolean {
```

Comment:
I think this whole method can be moved in FieldsUtils. 
Also, perhaps we could reuse some method already there to achieve the same result